CRM/Event/Form/Registration.php

```diff
@@ -893,6 +893,165 @@ protected function addParticipant(&$form, $contactID) {
     return $participant;
   }
 
+  /**
+   * Get the array of price field value IDs on the form that 'count' as
+   * full.
+   *
+   * The criteria for full is slightly confusing as it has an exclusion around
+   * select fields if they are the default - or something...
+   *
+   * @param array $field
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   */
+  protected function getOptionFullPriceFieldValues(array $field): array {
+    $optionFullIds = [];
+    foreach ($field['options'] ?? [] as &$option) {
+      if ($this->isOptionFullID($option, $field)) {
+        $optionFullIds[$option['id']] = $option['id'];
+      }
+    }
+    return $optionFullIds;
+  }
+
+  /**
+   * Is the option a 'full ID'.
+   *
+   * It is not clear why this is different to the is_full calculation
+   * but it is used in a less narrow context, around validation.
+   *
+   * Ideally figure it out & update this doc block.
+   *
+   * @param array $option
+   * @param array $field
+   *
+   * @return bool
+   * @throws \CRM_Core_Exception
+   */
+  private function isOptionFullID(array $option, array $field) : bool {
```

Comment:
Seems like we have a few different styles when it comes to return type hints. I usually put a single space between the `:` and the type but no space after the closing `)`. The above function does it that way but this function doesn't.
So for consistency:
```suggestion
  private function isOptionFullID(array $option, array $field): bool {
```